theseus_gui/src/components/ui/Instance.vue

```diff
@@ -11,23 +17,108 @@ const props = defineProps({
       return {}
     },
   },
+  load: {
+    type: Function,
+    default: () => {},
+  },
 })
+
+const confirmModal = ref(null)
+const stopBtn = ref(null)
+const playBtn = ref(null)
+
+const uuid = ref(null)
+
+const router = useRouter()
+
+const seeInstance = async () => {
+  const instancePath = props.instance.metadata
+    ? `/instance/${encodeURIComponent(props.instance.path)}`
+    : `/project/${encodeURIComponent(props.instance.project_id)}`
+
+  await router.push(instancePath)
+}
+
+const install = async (e) => {
+  e.stopPropagation()
+  props.load()
+  const [data, versions] = await Promise.all([
+    ofetch(
+      `https://api.modrinth.com/v2/project/${
+        props.instance.metadata
+          ? props.instance.metadata?.linked_project_id
+          : props.instance.project_id
+      }`
+    ).then(shallowRef),
+    ofetch(
+      `https://api.modrinth.com/v2/project/${
+        props.instance.metadata
+          ? props.instance.metadata?.linked_project_id
+          : props.instance.project_id
+      }/version`
+    ).then(shallowRef),
+  ])
+
+  if (data.value.project_type === 'modpack') {
+    const packs = Object.values(await list())
```

Comment:
you don't need to fetch a project and all of it's versions to determine the latest version ID and the project type, all of it is available in the search struct

Reply:
Closing after sync convo.